widget/list.go

```diff
@@ -93,6 +93,22 @@ func (l *List) MinSize() fyne.Size {
 	return l.BaseWidget.MinSize()
 }
 
+func (l *List) RefreshItem(id ListItemID) {
+	if l.scroller != nil {
+		l.BaseWidget.Refresh()
+		lo := l.scroller.Content.(*fyne.Container).Layout.(*listLayout)
+		visible := lo.visible
+		canvas := fyne.CurrentApp().Driver().CanvasForObject(lo.list)
+		var focused fyne.Focusable
+		if canvas != nil {
+			focused = canvas.Focused()
+		}
+		if item, ok := visible[id]; ok {
+			lo.setupListItem(item, id, focused == item)
+		}
+	}
```

Comment:
We prefer to keep the main code path not indented as it is a bit easier to read, i.e. you just put the if-statement on top:
```go
if l.scroller == nil {
    return
}
```